connectors/cli/connector.py

```diff
@@ -100,11 +100,17 @@ async def __create_connector(
                     index_name, language
                 )
 
-            api_key = await self.__create_api_key(index_name)
+            try:
+                api_key = await self.__create_api_key(index_name)
+                api_key_id = api_key["id"]
+                api_key_encoded = api_key["encoded"]
+            except:
```

Comment:
I changed this so that the logging is in `connector.py`, and we don't even try to create an API key if we have an API key. Let me know if what I did makes sense for you.

Reply:
Thanks! Yes, it makes sense to me. Especially, I like the idea of skipping the API key creation if we can't create it. However, I'd like to suggest some style/design comments: 

I want to keep all `@click` API calls in one place (`connectors_cli.py`) and keep the connectors-related logic separately. So I would move logging logic back to `connectors_cli.py`.  We can pass a variable like `skip_api_key: true|false` to `Connector.create` to skip `__create_api_key` when needed. 

I'm also not sure if we need to log this information every time we create a connector, maybe we can log only when the framework had to create a key but couldn't? 

I have a similar [PR](https://github.com/elastic/connectors/pull/2092/files#diff-0ecfa802c7463dd87de48892f4d7667806546360c871edc786641da417b6b76aR105-R107). The CLI will stop generating API keys for native connectors and for serverless.

Sorry for asking for a change again and let me know if I can help with it.

Reply:
Changed this in latest :)